internal/ethapi/api.go

```diff
@@ -492,6 +498,9 @@ func (s *PersonalAccountAPI) SignTransaction(ctx context.Context, args Transacti
 	if args.GasPrice == nil && (args.MaxFeePerGas == nil || args.MaxPriorityFeePerGas == nil) {
 		return nil, errors.New("missing gasPrice or maxFeePerGas/maxPriorityFeePerGas")
 	}
+	if args.BlobVersionedHashes != nil && args.MaxFeePerBlobGas == nil {
```

Comment:
should also have the reverse? where `args.BlobVersionedHashes == nil && args.MaxFeePerBlobGas != nil` ?

Reply:
perhaps 

```go
if args.IsEIP4844() && (args.BlobVersionedHashes == nil || args.MaxFeePerBlobGas == nil)
```